velox/functions/sparksql/ArrayRepeat.cpp

```diff
@@ -0,0 +1,196 @@
+    BufferPtr indices = allocateIndices(totalCount, pool);
+    BufferPtr sizes = allocateSizes(numRows, pool);
+    BufferPtr offsets = allocateOffsets(numRows, pool);
+    auto rawIndices = indices->asMutable<vector_size_t>();
+    auto rawSizes = sizes->asMutable<vector_size_t>();
+    auto rawOffsets = offsets->asMutable<vector_size_t>();
+
+    vector_size_t offset = 0;
+    context.applyToSelectedNoThrow(rows, [&](auto row) {
+      rawSizes[row] = count;
+      rawOffsets[row] = offset;
+      std::fill(rawIndices + offset, rawIndices + offset + count, row);
+      offset += count;
+    });
+
+    return std::make_shared<ArrayVector>(
+        pool,
+        outputType,
+        nullptr,
+        numRows,
+        offsets,
+        sizes,
+        BaseVector::wrapInDictionary(nullptr, indices, totalCount, args[0]));
+  }
+
+  VectorPtr applyNonConstant(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& outputType,
+      exec::EvalCtx& context) const {
+    exec::DecodedArgs decodedArgs(rows, args, context);
+    auto countDecoded = decodedArgs.at(1);
+    int32_t totalCount = 0;
+    // Directly throws user exception if invalid count is used.
+    context.applyToSelectedNoThrow(rows, [&](auto row) {
+      if (countDecoded->mayHaveNulls() && countDecoded->isNullAt(row)) {
+        return;
+      }
+      auto count = countDecoded->valueAt<int32_t>(row);
+      if (count < 0) {
+        // Negative count is like zero count, and empty array will be returned.
+        count = 0;
+      } else {
+        // The max limit is aligned with Spark.
+        VELOX_USER_CHECK_LE(
+            count,
+            kMaxCount,
+            "Count argument of array_repeat function must be less than or equal to (MAX_INT32 - 15)");
+      }
+      totalCount += count;
+    });
+
+    const auto numRows = rows.end();
+    auto pool = context.pool();
+
+    // Allocate new vector for nulls if necessary.
+    BufferPtr nulls;
+    uint64_t* rawNulls = nullptr;
+    if (countDecoded->mayHaveNulls()) {
+      nulls = allocateNulls(numRows, pool);
+      rawNulls = nulls->asMutable<uint64_t>();
+    }
+
+    // Allocate new vectors for indices, lengths and offsets.
+    BufferPtr indices = allocateIndices(totalCount, pool);
+    BufferPtr sizes = allocateSizes(numRows, pool);
+    BufferPtr offsets = allocateOffsets(numRows, pool);
+    auto rawIndices = indices->asMutable<vector_size_t>();
+    auto rawSizes = sizes->asMutable<vector_size_t>();
+    auto rawOffsets = offsets->asMutable<vector_size_t>();
+
+    vector_size_t offset = 0;
+    context.applyToSelectedNoThrow(rows, [&](auto row) {
+      if (rawNulls && countDecoded->isNullAt(row)) {
+        // Returns null if count argument is null.
+        // e.g. repeat('a', null) -> null
+        bits::setNull(rawNulls, row);
+        return;
+      }
+      auto count = countDecoded->valueAt<int32_t>(row);
```

Comment:
Notice that when evaluating TRY(array_repeat(...)) you'll end up processing rows with very large counts here. See logic around 'remainingRows' in velox/functions/prestosql/Map.cpp

Reply:
Hi @mbasmanova, thanks for your comment!
I found Spark directly fails with exception thrown when invalid count is found. In addition, Spark has no `try` expression (cc @rui-mo). If I understand correctly, `try(array_repeat())` should not be planned for Spark. How do you think?

Reply:
@PHILO-HE 

> Spark has no try expression

I see. Still, VectorFunction implementation must comply with TRY behavior as the same logic is used to suppress errors in other exceptions. For example, AND and OR may evaluate conjuncts in different order and therefore must suppress errors when evaluating individual conjuncts.

https://facebookincubator.github.io/velox/develop/expression-evaluation.html#adaptive-conjunct-reordering-in-and-and-or

Reply:
@mbasmanova, thanks for your comment! I just added `deselectErrors` to filter out rows with error in the new commit. Could you please help check again?